src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ReferencedModulesTests.cs

```diff
@@ -558,6 +558,110 @@ .maxstack  1
             }
         }
 
+        [WorkItem(1170032)]
+        [Fact]
+        public void DuplicateTypeInMscorlib()
+        {
+            var sourceConsole =
+@"namespace System
+{
+    public class Console
+    {
+    }
+}";
+            var sourceObjectModel =
+@"namespace System.Collections.ObjectModel
+{
+    public class ReadOnlyDictionary<K, V>
+    {
+    }
+}";
+            var source =
+@"class C
+{
+    static void Main()
+    {
+        var t = typeof(System.Console);
+        var o = (System.Collections.ObjectModel.ReadOnlyDictionary<object, object>)null;
+    }
+}";
+            var systemConsoleComp = CreateCompilationWithMscorlib(
+                sourceConsole,
+                options: TestOptions.DebugDll,
+                assemblyName: "System.Console");
+            var systemConsoleRef = systemConsoleComp.EmitToImageReference();
+            var systemObjectModelComp = CreateCompilationWithMscorlib(
+                sourceObjectModel,
+                options: TestOptions.DebugDll,
+                assemblyName: "System.ObjectModel");
+            var systemObjectModelRef = systemObjectModelComp.EmitToImageReference();
+            var identityObjectModel = systemObjectModelRef.GetAssemblyIdentity();
+
+            // At runtime System.Runtime.dll contract assembly is replaced
+            // by mscorlib.dll and System.Runtime.dll facade assemblies;
+            // System.Console.dll and System.ObjectModel.dll are not replaced.
+
+            // Test different ordering of modules containing duplicates:
+            // { System.Console, mscorlib } and { mscorlib, System.ObjectModel }.
+            var contractReferences = ImmutableArray.Create(systemConsoleRef, SystemRuntimePP7Ref, systemObjectModelRef);
+            var runtimeReferences = ImmutableArray.Create(systemConsoleRef, MscorlibFacadeRef, SystemRuntimeFacadeRef, systemObjectModelRef);
+
+            // Verify the compiler reports duplicate types with facade assemblies.
+            var compilation = CreateCompilation(
+                source,
+                references: runtimeReferences,
+                options: TestOptions.DebugDll,
+                assemblyName: ExpressionCompilerUtilities.GenerateUniqueName());
+            compilation.VerifyDiagnostics(
+                // (5,31): error CS0433: The type 'Console' exists in both 'System.Console, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' and 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
+                //         var t = typeof(System.Console);
+                Diagnostic(ErrorCode.ERR_SameFullNameAggAgg, "Console").WithArguments("System.Console, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null", "System.Console", "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089").WithLocation(5, 31),
```

Comment:
b77a5c561934e089 is the Ecma key which is used on the desktop framework but not CoreCLR. The CoreCLR uses the SL key.